src/test/java/org/folio/rest/api/ItemStorageTest.java

```diff
@@ -1935,6 +1940,19 @@ public void cannotPostSynchronousBatchWithoutStatus() {
     }
   }
 
+  @Test
+  public void synchronousBatchItemsShouldHaveStatusDateOnCreation() {
+    final JsonArray itemArray = threeItems();
+
+    assertThat(postSynchronousBatch(itemArray), statusCodeIs(HttpStatus.HTTP_CREATED));
+
+    JsonObject item = null;
+    for (int i = 0; i < itemArray.size(); i++) {
+      item = getById(itemArray.getJsonObject(i).getString("id")).getJson();
+      assertThat(item.getJsonObject("status").getString("date"), notNullValue());
+    }
```

Comment:
Personally, I would prefer to avoid `for` loops like this where possible. I'm aware that you were likely taking the style from the test above it.

What do you think of the following code as an alternative?

```suggestion
    JsonArrayHelper.toList(itemArray).forEach(itemInArray -> {
      final var fetchedItem = getById(itemInArray.getString("id")).getJson();

      assertThat(fetchedItem.getJsonObject("status").getString("date"), notNullValue());
    });
```